### DISCUSSION TITLE
Allow to run automatic updates without daily backups

### DISCUSSION THREAD

**Discussion by ghost** [nextcloud/all-in-one#1034] [Ideas] [answered]
Could be a checkbox that is enabled by default. would be helpful if backups are made from another application or from the server's hoster.

**Comment by szaimen** [comment]
This is already implemented

**Reply by gadric** [reply]
In 1.7.0?

**Comment by szaimen** [comment]
Sorry, I misread. This option will not be implemented.

Implemented is the ability for daily backups without automatic updates.

**Reply by ghost** [reply]
But why will it not be implemented too?

**Reply by szaimen** [reply]
Because we would risk breaking instances without having a backup in place. This is the only way to make sure that a backup is successfully created before running any update. 

However there is this: https://github.com/nextcloud/all-in-one#how-to-stopstartupdate-containers-or-trigger-the-daily-backup-from-a-script-externally

**Reply by ghost** [reply]
But I can create the time sequence so that the backup is created before the update. Also, the snapshot function of my rented server works as well.

**Reply by szaimen** [reply]
I doubt that you do the backup correctly since all containers should be stopped correctly before you create the backup which is only possible by following https://github.com/nextcloud/all-in-one#how-to-stopstartupdate-containers-or-trigger-the-daily-backup-from-a-script-externally. 

Anyways, my decision is final!

**Reply by ghost** [reply]
Can this be considered again? Because without this feature, when running on, for example, TrueNAS or another NAS operating system where existing data saves exist, duplicate backups of the data are made. But the function for automatic updates and the associated notifications are lost.

That would be a real pity. :(

**Reply by VasilisThePikachu** [reply]
> I doubt that you do the backup correctly since all containers should be stopped correctly before you create the backup which is only possible by following https://github.com/nextcloud/all-in-one#how-to-stopstartupdate-containers-or-trigger-the-daily-backup-from-a-script-externally.
> 
> Anyways, my decision is final!

Why not just give aio the ability to only do backups before upgrading? i dont see the issue with that. Since if the instance does break you already have a most recent backup of the previous version (and nextcloud does not have to shutdown every time for backups)

Yes backups are good but dont force them down peoples throats (especially since it sents a notification and email every time which gets annoying)

**Reply by ghost** [reply]
> > I doubt that you do the backup correctly since all containers should be stopped correctly before you create the backup which is only possible by following https://github.com/nextcloud/all-in-one#how-to-stopstartupdate-containers-or-trigger-the-daily-backup-from-a-script-externally.
> > Anyways, my decision is final!
> 
> Why not just give aio the ability to only do backups before upgrading? i dont see the issue with that. Since if the instance does break you already have a most recent backup of the previous version (and nextcloud does not have to shutdown every time for backups)
> 
> Yes backups are good but dont force them down peoples throats (especially since it sents a notification and email every time which gets annoying)

I agree... You could build the setting option in the AIO interface as follows:
![grafik](https://user-images.githubusercontent.com/90777102/202774942-deb23ba0-9bc0-410e-9763-41e53bca53af.png)
And if you don't want backups, you can uncheck "Activate this backup task".

@szaimen what do you think of this structure of the setting options?

**Comment by szaimen** [comment]
I'll think about maybe adding a config flag for this but it will definitely not be added as option to the AIO interface. Would this be sufficient for you?

**Reply by szaimen** [reply]
Ping people!

**Reply by ghost** [reply]
Yes I think that is sufficient.

**Reply by ghost** [reply]
And backups can be done cleanly by having the hypervisor shut down the virtual machine or docker and restart it after the backups.

**Reply by szaimen** [reply]
all right. Will think about a way to implement this option going forward.

**Comment by szaimen** [comment, marked as answer]
So I thought about this again and came to the conclusion that this is already possible and just needs a bit of documetation which was added with https://github.com/nextcloud/all-in-one/pull/1423.

Feel free to check out https://github.com/nextcloud/all-in-one#how-to-enable-automatic-updates-without-creating-a-backup-beforehand

**Reply by szaimen** [reply]
Just as a heads-up: this seems to not work completely correctly yet as there might be some race condition happening. This will get fixed with https://github.com/nextcloud/all-in-one/pull/1428 and the next release

**Reply by ghost** [reply]
All right thanks for the heads up and for the fix.

**Reply by szaimen** [reply]
This is now released with v.3.1.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
